server/GroupEmote.lua

```diff
@@ -0,0 +1,110 @@
+local groupEmoteRequests = {
+    --[[
+    [reqID] = {
+        emote = "emote",
+        requestorID = 0,
+        requestorName = "PlayerName",
+        zoneOrigin = vector3,
+        zoneRadius = 5.0,
+        players = {
+            [requestorSrc] = true,
+            [src] = false,
+            [src2] = true,
+            ...
+        }
+    }
+    ]]--
+}
+
+function CreateGroupEmoteRequest(reqid, emote, source)
+    return {
+        emote = emote,
+        requestorID = source,
+        requestorName = GetPlayerName(source) or "INVALID",
+        zoneOrigin = GetEntityCoords(GetPlayerPed(source)), -- Coords where the group request was started.
+        zoneRadius = 5.0,
+        players = {}
+    }
+end
+
+RegisterNetEvent("rpemotes:server:startGroupEmote", function(emote, players)
+    local source = source
+    -- We will send the list of players from the client.
+    -- We could loop through all the players server-side, but I am not sure servers with ~1000 players online would appreciate it.
+    -- We'll sanitize the data received from the player instead.
+
+    if not Player(source).state.canEmote then return end
+    if type(players) ~= "table" or #players == 0 then return end
+    
+    -- Create groupEmoteRequest
+    ::retry_reqid::
+    local reqid = tostring(os.time())..tostring(source)..tostring(math.random(0,99))
+    if groupEmoteRequests[reqid] then goto retry_reqid end
+    groupEmoteRequests[reqid] = CreateGroupEmoteRequest(reqid, emote, source)
+
+    local zone = {
+        coords = groupEmoteRequests[reqid].zoneOrigin,
+        radius = groupEmoteRequests[reqid].zoneRadius
+    }
+
+    -- Add Players to the group request
+    local sourceCoords = GetEntityCoords(GetPlayerPed(source))
+    for i=1, 128, 1 do -- Max number of players that can be in one scope.
+        if not players[i] or type(players[i]) ~= "number" then
+            break
+        end
+        if GetPlayerPing(players[i]) ~= 0 then
+            local src = players[i]
+            local ped = GetPlayerPed(src)
+            local crds = DoesEntityExist(ped) and GetEntityCoords(ped)
+            if #(sourceCoords - crds) <= groupEmoteRequests[reqid].zoneRadius then
+                groupEmoteRequests[reqid].players[src] = false
```

Comment:
Maybe this should be nil instead of false? You'll be able to simplify some later code to avoid checking for a true value, iterating over the players will be enough to confirm they are in the set.